sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -161,84 +200,204 @@ private boolean methodIsIneligible(MethodVertex vertex) {
+        // Start with each UserClassVertex for the defining types.
+        List<UserClassVertex> subclassVertices =
                 SFVertexFactory.loadVertices(
                         g,
-                        g.V().or(
-                                        __.where(
-                                                        H.has(
-                                                                NodeType.METHOD_CALL_EXPRESSION,
-                                                                Schema.DEFINING_TYPE,
-                                                                definingType))
-                                                // NOTE: There's no .hasLabel(String...) method,
-                                                // hence the call to
-                                                // .hasLabel(String,String...) with the array
-                                                // defined earlier.
-                                                .where(
-                                                        __.repeat(__.out(Schema.CHILD))
-                                                                .until(
-                                                                        __.hasLabel(
-                                                                                desiredReferenceTypes[
-                                                                                        0],
-                                                                                desiredReferenceTypes))
-                                                                .count()
-                                                                .is(P.gt(0))),
-                                        __.where(
-                                                H.has(
-                                                        NodeType.THIS_METHOD_CALL_EXPRESSION,
-                                                        Schema.DEFINING_TYPE,
-                                                        definingType))));
-
-        // Cache the result, then return it.
-        this.internalMethodCallsByDefiningType.put(definingType, internalCalls);
-        return internalCalls;
+                        g.V().where(
+                                        H.hasWithin(
+                                                NodeType.USER_CLASS,
+                                                Schema.DEFINING_TYPE,
+                                                definingTypes))
+                                // Go out to every subclass.
+                                .out(Schema.EXTENDED_BY));
+        return subclassVertices.stream().map(UserClassVertex::getName).collect(Collectors.toList());
+    }
+
+    /**
+     * Applies the provided validator to each of the provided method call expressions, to see if
+     * they are an invocation of the desired method.
+     */
+    private boolean validatorDetectsUsage(
```

Comment:
Moved the guts of the validation invocation into this method, so that each form of validation can be invoked in the same way.